src/cart/crt_internal_types.h

```diff
@@ -166,6 +166,160 @@ struct crt_event_cb_priv {
+	ENV_STR(D_DOMAIN)                                                                          \
+	ENV_STR(D_FI_CONFIG)                                                                       \
+	ENV_STR(D_INTERFACE)                                                                       \
+	ENV_STR(D_LOG_FILE)                                                                        \
+	ENV_STR(D_LOG_FILE_APPEND_PID)                                                             \
+	ENV_STR(D_LOG_FILE_APPEND_RANK)                                                            \
+	ENV_STR(D_LOG_FLUSH)                                                                       \
+	ENV_STR(D_LOG_MASK)                                                                        \
+	ENV_STR(D_LOG_SIZE)                                                                        \
+	ENV(D_LOG_STDERR_IN_LOG)                                                                   \
+	ENV(D_POLL_TIMEOUT)                                                                        \
+	ENV_STR(D_PORT)                                                                            \
+	ENV(D_PORT_AUTO_ADJUST)                                                                    \
+	ENV(D_POST_INCR)                                                                           \
+	ENV(D_POST_INIT)                                                                           \
+	ENV_STR(D_PROVIDER)                                                                        \
+	ENV_STR_NO_PRINT(D_PROVIDER_AUTH_KEY)                                                      \
+	ENV(D_QUOTA_RPCS)                                                                          \
+	ENV(FI_OFI_RXM_USE_SRX)                                                                    \
+	ENV(FI_UNIVERSE_SIZE)                                                                      \
+	ENV(SWIM_PING_TIMEOUT)                                                                     \
+	ENV(SWIM_PROTOCOL_PERIOD_LEN)                                                              \
+	ENV(SWIM_SUSPECT_TIMEOUT)                                                                  \
+	ENV_STR(UCX_IB_FORK_INIT)
+
+/* uint env */
+#define ENV(x)                                                                                     \
+	unsigned int _##x;                                                                         \
+	int          _rc_##x;                                                                      \
+	int          _no_print_##x;
+
+/* char* env */
+#define ENV_STR(x)                                                                                 \
+	char *_##x;                                                                                \
+	int   _rc_##x;                                                                             \
+	int   _no_print_##x;
+
+#define ENV_STR_NO_PRINT(x) ENV_STR(x)
+
+struct crt_envs_t {
```

Comment:
we should not not use `_t` after struct names :)

Reply:
ok